common/src/main/scala/uk/ac/wellcome/sqs/SQSReader.scala

```diff
@@ -24,75 +24,77 @@ class SQSReader @Inject()(sqsClient: AmazonSQS, sqsConfig: SQSConfig)
   // If the timeout expires before the consumer sends a delete request, the message is unhidden and can be read by another consumer.
 
   def retrieveAndDeleteMessages(
-    process: Message => Future[Unit]): Future[Unit] =
-    Future {
-      blocking {
-        debug(s"Looking for new messages at ${sqsConfig.queueUrl}")
-        receiveMessages()
-      }
-    } flatMap { messages =>
-      if (messages.nonEmpty)
-        info(
-          s"Received messages ${messages.map { _.getMessageId }} from queue ${sqsConfig.queueUrl}")
-      else
-        debug(
-          s"Received messages ${messages.map { _.getMessageId }} from queue ${sqsConfig.queueUrl}")
-      processAndDeleteMessages(messages, process).map { _ =>
+    process: Message => Future[Unit]): Future[Unit] = {
+    val eventuallyProcessMessages =
+      for {
+        _ <- Future.successful {
+          debug(s"Looking for new messages at ${sqsConfig.queueUrl}")
+        }
+        ms <- receiveMessages()
+        _ <- Future.sequence { ms.map { processAndDeleteMessage(_, process) } }
+      } yield {
+        val messageIds = ms.map { _.getMessageId }
+
+        if (ms.nonEmpty) {
+          info(
+            s"Received messages $messageIds from queue ${sqsConfig.queueUrl}")
+        } else {
+          debug(
+            s"Received messages $messageIds from queue ${sqsConfig.queueUrl}")
```

Comment:
Would it be slightly cheaper to do

```scala
debug(s"Received no messages from queue ${sqsConfig.queueUrl}")
```

Save us the cost of a string interpolation, and makes it stand out more from the previous message.